### PR TITLE
Add effective url options to slice plugin.

### PR DESCRIPTION
**Pull request by traeak**
Initial slice plugin used pristine url to virtually call back into the same rule.  This allows using a single remap line, but it has a disadvantage of rerunning all plugin rules for each slice transaction in addition to the incoming request transaction.  Also it makes log analysis more difficult, the only difference being that the callback comes in on port '0'.  http and https get a bit confusing as well with using the pristine url as the loopback is connected with the https url if the incoming rule is https.

This changes the default remap behaviour of the slice plugin to use a virtual remap parent point, allowing separate remap lines.

the way to set up the slice plugin with effective url: (option --remap-host=<loop name> or -r <loop name>)

```
# Base remap rule (non GET requests pass through)
map http://ats/ http://parent/ @plugin=uri_signing.so @pparam=foo_uri.config @plugin=slice.so @pparam=--remap-host=loopback

# Virtual forward proxy slice block leg (cached)
map http://loopback/ http://parent/ @plugin=header_rewrite.so @pparam=reject_non_php0.config @plugin=cache_range_requests.so
```

Previous configuration is still default behavior (uri_signs for incoming AND each sliced up block):

```
map http://ats/ http://parent/ @plugin=uri_signing.so @pparam=foo_uri.config @plugin=slice.so @plugin=cache_range_requests.so
```

### CODE REVIEW

**Comment by traeak** [comment]
Dealing with head requests for the effective url path.

**Comment by bryancall** [comment]
[approve ci]

File: doc/admin-guide/plugins/slice.en.rst [hunk 1]
```diff
@@ -204,14 +214,43 @@ by cache_range_requests.  The parent will trim those requests to
...
+check for the existence of any headers they may have created the first
+time they have were visited.
+
+To get around this the '--remap-host=<host>' or '-r <host>' option may
+be used.  This requires an intermediate loopback remap to be defined which
+handles each slice block request.
+
+This works well with any remap rules that use the url_sig or uri_signing
+plugins.  As the client remap rule is not caching any plugins that
+manipulate the cache key would need to go into the loopback to parrent
```
Comment (jrushford) [inline doc/admin-guide/plugins/slice.en.rst]: parret -> parent

Comment (traeak) [reply doc/admin-guide/plugins/slice.en.rst]: typo fix!

**Comment by jrushford** [comment]
[approve ci ubuntu]

**Comment by jrushford** [comment]
[approve ci ubuntu]

**Comment by zwoop** [comment]
Do you want this for 9.0.x ?

**Comment by jrushford** [comment]
Yes please.

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**